src/sb.c

```diff
@@ -26,15 +26,14 @@ bloom_hashval bloom_calc_hash64(const void *buffer, int len);
 #define CUR_FILTER(sb) ((sb)->filters + ((sb)->nfilters - 1))
 
 static int SBChain_AddLink(SBChain *chain, uint64_t size, double error_rate) {
-    if (!chain->filters) {
-        chain->filters = RedisModule_Calloc(1, sizeof(*chain->filters));
-    } else {
-        chain->filters =
-            RedisModule_Realloc(chain->filters, sizeof(*chain->filters) * (chain->nfilters + 1));
-    }
+    chain->filters =
+        RedisModule_Realloc(chain->filters, sizeof(*chain->filters) * (chain->nfilters + 1));
```

Comment:
It would change the allocation size when filters=0?
Is it intentional?

Reply:
It will still allocate same amount of memory. When `chain->filters` is NULL, `chain->nfilters` is zero. So total allocation is `sizeof(*chain->filters) * 1`.